document-generator-core/src/main/java/uk/gov/companieshouse/document/generator/core/service/impl/DocumentGeneratorServiceImpl.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.companieshouse.document.generator.core.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import uk.gov.companieshouse.document.generator.core.document.render.RenderDocumentRequestHandler;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentRequest;
+import uk.gov.companieshouse.document.generator.core.document.render.models.RenderDocumentResponse;
+import uk.gov.companieshouse.document.generator.core.service.DocumentGeneratorService;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentRequest;
+import uk.gov.companieshouse.document.generator.core.service.models.DocumentResponse;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfo;
+import uk.gov.companieshouse.environment.EnvironmentReader;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import static uk.gov.companieshouse.document.generator.core.DocumentGeneratorApplication.APPLICATION_NAME_SPACE;
+
+public class DocumentGeneratorServiceImpl implements DocumentGeneratorService {
+
+    private DocumentInfoService documentInfoService;
+
+    private EnvironmentReader environmentReader;
```

Comment:
If a static EnvironmentReaderImpl is instantiated at the class level, we can have static Strings on the class to hold the environment variable details; that way, we don't end up reading env files on every method call. Something like:

```
public class Class {

   private static final EnvironmentReader READER = new EnvironmentReaderImpl();

   private static String DOCUMENT_RENDER_SERVICE_HOST = READER.getMandatoryString("DOCUMENT_RENDER_SERVICE_HOST");

}
```

The added benefit of this is that any `MissingEnvironmentVariableException`'s will be thrown on class instantiation, as opposed to when methods are called so we fail earlier.

Reply:
problem with that is you can't mock a static variable. So that is why we are doing construction injection which is a Spring recommendation.

Reply:
Fair point, disregard this one then